### ISSUE TITLE
Support createBulk C8Y JSON topics

### ISSUE THREAD

**Issue by zhongys-c8y** [thin-edge/thin-edge.io#3203]
**Issue description**
A customer wants to send bulk measurements via thin-edge, each measurement has different timestamps. The current thin-edge JSON format can include timestamp only on the top level. But C8Y JSON supports createBulk. 

Then I found that thin-edge dosen't support createBulk topics out-of-the-box. 

**Workaround**
User can still use createBulk topics by:
1.  Add the topic into c8y-bridge.conf, for example: `topic measurement/measurements/createBulk out 2 c8y/ ""`
2.  Restart mosquitto service: `systemctl restart mosquitto.service`
Then thin-edge can forward the measurements to the platform. But thin-edge will lose this custom config after reconnecting tedge. 

**Expecting  solution**
It would be great if thin-edge can support createBulk topics natively. According to C8Y [documentation](https://cumulocity.com/docs/smartrest/json-via-mqtt/#supported-endpoint), C8Y supports:
` event/events/createBulk`
` alarm/alarms/createBulk`
` measurement/measurements/createBulk`

**Comment by reubenmiller** [comment]
Note: During development each of the added endpoints should be checked if they do work as intended (e.g. if they do create the telemetry data in Cumulocity). There is the possibility that the Cumulocity docs are not 100% correct, as there is some doubt on the functionality behind the `alarm/alarms/createBulk` and `event/events/createBulk` topics.

**Comment by reubenmiller** [comment]
I can confirm that all 3 `createBulk` topics function as documented. The topics were tested by manually modifying the `c8y-bridge.conf` mosquitto bridge configuration and using the following commands:

## event/events/createBulk

```sh
tedge mqtt pub c8y/event/events/createBulk "{
    \"events\": [
        {\"text\":\"bulk event 1\",\"type\":\"type1\",\"time\":\"$(date -Iseconds)\"},
        {\"text\":\"bulk event 2\",\"type\":\"type2\",\"time\":\"$(date -Iseconds)\"}
    ]
}"
```

**child device**

```sh
tedge mqtt pub c8y/event/events/createBulk "{
    \"events\": [
        {\"text\":\"bulk event 1\",\"type\":\"type1\",\"time\":\"$(date -Iseconds)\",\"externalSource\":{\"externalId\":\"issue3234:device:child01\",\"type\":\"c8y_Serial\"}},
        {\"text\":\"bulk event 2\",\"type\":\"type2\",\"time\":\"$(date -Iseconds)\",\"externalSource\":{\"externalId\":\"issue3234:device:child01\",\"type\":\"c8y_Serial\"}}
    ]
}"
```

```sh
tedge mqtt pub c8y/event/events/createBulk "{
    \"events\": [
        {\"text\":\"bulk event 1\",\"type\":\"example\",\"time\":\"2024-12-01T02:00:00Z\",\"externalSource\":{\"externalId\":\"issue3234:device:child01\",\"type\":\"c8y_Serial\"}},
        {\"text\":\"bulk event 2\",\"type\":\"example\",\"time\":\"2024-12-01T02:01:00Z\",\"externalSource\":{\"externalId\":\"issue3234:device:child01\",\"type\":\"c8y_Serial\"}}
    ]
}"
```

## alarm/alarms/createBulk

```sh
tedge mqtt pub c8y/alarm/alarms/createBulk "{
    \"alarms\": [
        {\"text\":\"bulk alarm 1\",\"type\":\"type1\",\"severity\":\"MAJOR\",\"time\":\"$(date -Iseconds)\"},
        {\"text\":\"bulk alarm 2\",\"type\":\"type2\",\"severity\":\"MAJOR\",\"time\":\"$(date -Iseconds)\"}
    ]
}"
```

**child device**

```sh
tedge mqtt pub c8y/alarm/alarms/createBulk "{
    \"alarms\": [
        {\"text\":\"bulk alarm 1\",\"type\":\"type1\",\"severity\":\"MAJOR\",\"time\":\"$(date -Iseconds)\",\"externalSource\":{\"externalId\":\"issue3234:device:child01\",\"type\":\"c8y_Serial\"}},
        {\"text\":\"bulk alarm 2\",\"type\":\"type2\",\"severity\":\"MAJOR\",\"time\":\"$(date -Iseconds)\",\"externalSource\":{\"externalId\":\"issue3234:device:child01\",\"type\":\"c8y_Serial\"}}
    ]
}"
```

## measurement/measurements/createBulk

```sh
tedge mqtt pub c8y/measurement/measurements/createBulk "{
    \"measurements\": [
        {\"env\":{\"temperature\":{\"value\":20.3,\"unit\":\"degC\"}},\"time\":\"$(date -Iseconds)\",\"type\":\"10_min\"},
        {\"env\":{\"temperature\":{\"value\":20.3,\"unit\":\"degC\"}},\"time\":\"$(date -Iseconds)\",\"type\":\"10_min\"}
    ]
}"
```

**child device**

```sh
tedge mqtt pub c8y/measurement/measurements/createBulk "{
    \"measurements\": [
        {\"env\":{\"temperature\":{\"value\":20.3,\"unit\":\"degC\"}},\"time\":\"$(date -Iseconds)\",\"type\":\"10_min\",\"externalSource\":{\"externalId\":\"issue3234:device:child01\",\"type\":\"c8y_Serial\"}},
        {\"env\":{\"temperature\":{\"value\":20.3,\"unit\":\"degC\"}},\"time\":\"$(date -Iseconds)\",\"type\":\"10_min\",\"externalSource\":{\"externalId\":\"issue3234:device:child01\",\"type\":\"c8y_Serial\"}}
    ]
}"
```

**Comment by reubenmiller** [comment]
It's been merged and will be included in the [next release 1.4.0](https://github.com/thin-edge/thin-edge.io/milestone/33).

But if you want to try it out now, you can install thin-edge.io from the main channel, e.g.

```sh
wget -O - thin-edge.io/install.sh | sh -s -- --channel main
```
